src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,36 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder builder = new StringBuilder("Report for period " + dateFrom + " - " + dateTo);
+
+        for (String name : names) {
+            int totalSalary = 0;
+            for (String dates : data) {
+                String[] employeeInfo = dates.split(" ");
+                if (employeeInfo[1].equals(name)) {
+                    if (isDateTheSameOrBetween(employeeInfo[0], dateFrom, dateTo)) {
+                        totalSalary += Integer.parseInt(employeeInfo[2])
+                                * Integer.parseInt(employeeInfo[3]);
```

Comment:
it's better to extract hardcoded indexes to constants with proper naming to increase code readability and maintainability, like 
```suggestion
                if (employeeInfo[NAME_INDEX].equals(name)) {
                    if (isDateTheSameOrBetween(employeeInfo[DATE_INDEX], dateFrom, dateTo)) {
                        totalSalary += Integer.parseInt(employeeInfo[WORKED_HOURS_INDEX])
                                * Integer.parseInt(employeeInfo[PER_HOUR_SALARY_INDEX]);
```

Reply:
I have a question
If index uses just once (as my indexes 0,1,2,3) we should move them to constants? Index[0] uses just one in code, the rest as well

Reply:
I'd say it depends. Here, the name that reflects the role of a variable (NAME_INDEX etc.) significantly increases readability - it's immediately apparent what's going on in the code, especially to the person who will read it for the first time - NAME_INDEX is more informative than 1, it makes reading the code line feel like English text. It's not always about reducing duplication, even if a value is used once it's often worth it to extract it to a constant

Reply:
It also makes it easier to make changes - if the format of input data changes, it's easier to make amends to constant at the top of the file, than to search for the hardcoded literals in the file